pkg/handler3/handler.go

```diff
@@ -191,6 +191,7 @@ func ToV3ProtoBinary(json []byte) ([]byte, error) {
 
 func (o *OpenAPIService) HandleDiscovery(w http.ResponseWriter, r *http.Request) {
 	data, _ := o.getGroupBytes()
+	w.Header().Set("Etag", strconv.Quote(computeETag(data)))
```

Comment:
Do we need to change/set some cache headers too? I think we only accept `json`, should we set the content-type? Should we reject requests with non-`application/json` accept headers? Since we only have one content-type, I guess we probably don't need `Vary`?

Reply:
Updated. Unsure about rejecting non-"application/json" requests since we have a few cases that might also need to be checked for `*/*`, `application/*`, */json`, empty.